### PR TITLE
[workflows] Add workflows for buliding and testing

### PR DESCRIPTION
**Pull request by WonyoungChoi**
This PR is to add github actions workflows for building the engine source and running unit-tests.

## `Build` workflow is :
- using GitHub's hosted runner.
- using `actions/cache` for incremental build, the cache size for each job is about 140~200MB. 
- using `strategy matrix ` to create multiple jobs in building task.
- using the docker image `build-engine`.

## `Check Symbols` workflow :
- checks symbols of the build artifacts with the allowlist. 
- is triggered by `workflow_run` event. (for security reason)

## tasks
- [x] Build engines for each target [arm-debug, arm-profile, arm-release, arm64-debug, arm64-profile, arm64-release, x86-debug]
- [x] Run unit tests
- [x] Verify symbol references.
- [x] Publish artifacts.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+      - name: gclient sync
+        run: |
+          gclient-prepare-sync.sh --reduce-deps --shallow-sync
+          gclient sync -v --no-history --shallow
+
+      - name: check format
+        if: matrix.os == 'host'
+        run: |
+          src/flutter/ci/format.sh
```
Comment (WonyoungChoi) [inline .github/workflows/build.yml:76]: this step will be removed by https://github.com/flutter-tizen/engine/pull/188

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            --runtime-mode ${{ matrix.mode }} $BUILD_TARGET_OPT
+          cache-checksum.sh save src/out/$OUTPUT_NAME
+
+      - uses: actions/upload-artifact@v2
+        if: matrix.os == 'host'
+        with:
+          name: host-${{ matrix.arch }}-${{ matrix.mode }}
+          path: |
+            src/out/${{ env.OUTPUT_NAME }}/libflutter_*.so
+            src/out/${{ env.OUTPUT_NAME }}/*_unittests
```
Comment (swift-kim) [inline .github/workflows/build.yml:86]: The size of `libflutter_*.so` (`libflutter_engine.so` in this case) is too large (over 100 MB). As far as I know the file is useless (nothing depends on the file) so you could either remove the line (if you just want to run unit tests) or include `flutter_tizen_shell` (executable) instead.

File: .github/workflows/check-symbol.yml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    workflows: ["Build"]
+    types:
+      - requested
+      - completed
+
+jobs:
+  pending:
+    if: |
+      github.event.action == 'requested' &&
+      github.event.workflow_run.event == 'pull_request'
```
Comment (swift-kim) [inline .github/workflows/check-symbol.yml:14]: Why do we need this job?

Comment (WonyoungChoi) [reply .github/workflows/check-symbol.yml:14]: This is a job to display the "Check Symbols" checker status as below when build starts.
image

File: .github/workflows/build.yml (hunk 2 above)
Comment (WonyoungChoi) [reply .github/workflows/build.yml:86]: Thanks for the information. I will remove the line for `libflutter-*.so`.

File: .github/workflows/check-symbol.yml (hunk 3 above)
Comment (swift-kim) [reply .github/workflows/check-symbol.yml:14]: Hmm. But, why? Those three steps add extra complexity to the workflow file and the information doesn't seem very useful. Won't `check-symbol.py` exit with non-zero exit code if anything goes wrong?

Also, how does separating the Check Symbols workflow from the Build workflow improve security? Is it different from adding a job that depends on the build job in the Build workflow?

Comment (WonyoungChoi) [reply .github/workflows/check-symbol.yml:14]: 1. I agree, `pending` job is not necessary. But without this job, the "Check Symbol" checker will be shown in the PR status only after the build is over.
2. To check the symbol, the workflow have to access the allowlist in private repo with the secret `TIZENAPI_TOKEN'. However, for security reasons, these secret keys are not accessible in the forked pull request event workflows. 
- https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
- https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#workflow_run

Comment (swift-kim) [reply .github/workflows/check-symbol.yml:14]: > But without this job, the "Check Symbol" checker will be shown in the PR status only after the build is over.

That's okay. I don't see any problem with it.

> However, for security reasons, these secret keys are not accessible in the forked pull request event workflows.

Okay. Now I understand that it's one of GitHub's limitations. I have a slightly different idea but I'll talk to you offline later.

**Review by swift-kim — APPROVED** [review]

**Pull request merged**